### ISSUE TITLE
User Message: failed to execute command /var/lib/teleport/backend error:  too many open files

### ISSUE THREAD

**Issue by andrewbanchich** [gravitational/teleport#2513]
I've been having a lot of issues with Teleport for the last couple weeks. When I try to log in to the Web UI, the OTP token field will sometimes be missing, or it will be there but take forever to load and then say "Bad Auth". When I log in to the server and run `sudo journalctl -fu teleport` it is filled with logs like this:

`Jan 21 19:39:38 teleport[22527]: ERRO [AUTH]      "failed to retrieve client pool: \nERROR REPORT:\nOriginal Error: *trace.AccessDeniedError failed to execute command /var/lib/teleport/backend error:  too many open files\nStack Trace:\n\t/gopath/src/github.com/gravitational/teleport/lib/backend/dir/impl.go:142 github.com/gravitational/teleport/lib/backend/dir.(*Backend).GetItems\n\t/gopath/src/github.com/gravitational/teleport/lib/backend/sanitize.go:78 github.com/gravitational/teleport/lib/backend.(*Sanitizer).GetItems\n\t/gopath/src/github.com/gravitational/teleport/lib/services/local/trust.go:206 github.com/gravitational/teleport/lib/services/local.(*CA).GetCertAuthorities\n\t/gopath/src/github.com/gravitational/teleport/lib/auth/auth.go:666 github.com/gravitational/teleport/lib/auth.(*AuthServer).ClientCertPool\n\t/gopath/src/github.com/gravitational/teleport/lib/auth/middleware.go:129 github.com/gravitational/teleport/lib/auth.(*TLSServer).GetConfigForClient\n\t/gopath/src/github.com/gravitational/teleport/lib/auth/middleware.go:110 github.com/gravitational/teleport/lib/auth.(*TLSServer).GetConfigForClient-fm\n\t/opt/go/src/crypto/tls/handshake_server.go:128 crypto/tls.(*serverHandshakeState).readClientHello\n\t/opt/go/src/crypto/tls/handshake_server.go:48 crypto/tls.(*Conn).serverHandshake\n\t/opt/go/src/crypto/tls/conn.go:1309 crypto/tls.(*Conn).Handshake\n\t/opt/go/src/net/http/server.go:1713 net/http.(*conn).serve\n\t/opt/go/src/runtime/asm_amd64.s:2338 runtime.goexit\nUser Message: failed to execute command /var/lib/teleport/backend error:  too many open files\n" auth/middleware.go:133`

I'm not really sure what could be causing this. Any ideas?

**Environment**:
- Teleport version (use `teleport version`): Teleport v3.0.0 git:v3.0.0-0-g017ae5f9
- Tsh version (use `tsh version`):Teleport v3.0.0 git:v3.0.0-0-g017ae5f9
- OS (e.g. from /etc/os-release): Ubuntu 18.04

**Comment by klizhentas** [comment]
It seems like Teleport opened too many files, can you try `lsof -p $(pidof teleport)`

**Comment by andrewbanchich** [comment]
`teleport 23053 root  cwd   unknown                      /proc/23053/cwd (readlink: Permission denied)
teleport 23053 root  rtd   unknown                      /proc/23053/root (readlink: Permission denied)
teleport 23053 root  txt   unknown                      /proc/23053/exe (readlink: Permission denied)
teleport 23053 root NOFD                                /proc/23053/fd (opendir: Permission denied)`

**Comment by andrewbanchich** [comment]
If I `lsof | grep teleport` there is a ton of them though.

**Comment by klizhentas** [comment]
Let me rephrase, can you run an appropriate command/set of commands to:

* List total amount of files opened by teleport
* Group them by file/socket type
* Anonymize it, and paste it here

**Comment by andrewbanchich** [comment]
@klizhentas Sure, what (other than IP addresses) should I look for to strip out to anonymize this before posting?

**Comment by andrewbanchich** [comment]
Hmm, I must have not run the correct command you asked for. Sorry! This command prints out a wall of info with a lot of IP addresses / domains in it. I'm not sure how long it would take me to strip it out. Is there a way I can send it to you privately?

**Comment by klizhentas** [comment]
Sure, join [Community Slack](https://join.slack.com/t/grv8/shared_invite/enQtNTEwMTU2NjQ2NzU2LWIzYTM1ZDcyNWFhMzZhNDJhN2IyOWMwZDA4NDA4MmE2MjgzYWNiYmY1NjcxYjA3M2RmOTQ1ZjRkOTAwODE1NDY) and DM me there -  `sasha`

**Comment by klizhentas** [comment]
this was a perf issue related to the slowness of teleport on many trusted clusters. I recommend checking 4.0 post beta
